src/ray/raylet/worker_pool.cc

```diff
@@ -571,6 +593,15 @@ void WorkerPool::HandleJobFinished(const JobID &job_id) {
   // Currently we don't erase the job from `all_jobs_` , as a workaround for
   // https://github.com/ray-project/ray/issues/11437.
   // unfinished_jobs_.erase(job_id);
+  auto job_config = GetJobConfig(job_id);
+  RAY_CHECK(job_config);
+  if (job_config->has_runtime_env_info() &&
+      job_config->runtime_env_info().runtime_env_eager_install()) {
```

Comment:
why don't we need the same check in HandleJobStarted?

Reply:
![image](https://user-images.githubusercontent.com/26714159/144538795-f38480d3-2ef4-49e0-84ba-b7c61f2687c6.png)
We already have this check?